lib/inets/src/http_server/httpd_file.erl

```diff
@@ -35,13 +35,17 @@ handle_error(enotdir, Op, ModData, Path) ->
 	         ": A component of the file name is not a directory");
 handle_error(eisdir, Op, ModData, Path) ->
     handle_error(403, Op, ModData, Path,
-	         ":Illegal operation expected a file not a directory");
+	         ": Illegal operation expected a file not a directory");
+handle_error(enametoolong, Op, ModData, Path) ->
+    handle_error(422, Op, ModData, Path,
```

Comment:
Maybe 414?

Reply:
No this will not work. The URI length is a separate check and it can be ok and still end up with this case.  Maybe mostly in malicious DoS attacks scenarios but need to take care of those too.

Reply:
I think 422 fits as in this case the client request is wrong, as it has been interpreted as a filename which it can not be as it is too long.

Reply:
From RFC 9110:

> The 422 (Unprocessable Content) status code indicates that the server understands the content type of the request content (hence a 415 (Unsupported Media Type) status code is inappropriate), and the syntax of the request content is correct, but it was unable to process the contained instructions. For example, this status code can be sent if an XML request content contains well-formed (i.e., syntactically correct), but semantically erroneous XML instructions.

Is the filename part of the request content?

> HTTP messages often transfer a complete or partial representation as the message "content": a stream of octets sent after the header section, as delineated by the message framing.

Reply:
Humm I think I read an older defenition, how about 404!

Reply:
I think 404 is much more appropriate, at least from my understanding of the http rfcs.